aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -67,23 +111,43 @@ sub get_anaconda_version
     return $version;
 }
 
-# Returns the absolute path where the PXE file must be written.
+# Retuns the IP-based PXE file name, based on the PXE variant
+sub hexip_filename
+{
+    my ($ip, $variant) = @_;
+
+    my $hexip_str = '';
```

Comment:
Why? The reason for doing this is that there is in general no check of the return value that tends to be used directly in string concatenation. This will raise a warning when this code ensures that the value can be used. Else, if we use undef, we need to check the return value on any usage. Not necessarily a bad idea but may be for a future version...